File: CPP/Implementation1_thread_unsafe/position.cpp
#include "position.h"
position::position()
{}
position::position(int x,int y)
:x_position(x),y_position(y)
{}

position::position(const position& p)
{
	x_position = p.x_position;
	y_position = p.y_position;
}

std::ostream& operator<<(std::ostream& o,const position& p)
{
	return (o << "X = " << p.x_position << " ; Y = " << p.y_position);
}

File: JAVA/Implementation1/Position.java
public class Position {
	
	public int xPosition;
	public int yPosition;

	public Position(int dx, int dy) {
		xPosition = dx;
		yPosition = dy;
	}
	
	@Override
	public String toString()
	{
		return "X = "+xPosition+" ; Y = "+yPosition;
		
	}
}